components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1463,6 +1474,60 @@ def get_thumb_url(iid):
 
 @login_required()
 @jsonp
+def get_thumbnails_json(request, w=None, conn=None, **kwargs):
+    """
+    Returns base64 encoded jpeg with the rendered thumbnail for images
+    'id'
+
+    @param request:     http request
+    @param w:           Thumbnail max width. 96 by default
+    @return:            http response containing base64 encoded thumbnails
+    """
+    if w is None:
+        w = 96
+    image_ids = get_longs(request, 'id')
+    logger.debug("Image ids: %r" % image_ids)
+    if len(image_ids) > settings.THUMBNAILS_BATCH:
+        return HttpJavascriptResponseServerError(
+            'Max %s thumbnails at a time.' % settings.THUMBNAILS_BATCH)
```

Comment:
> Is the primary reason for the THUMBNAILS_BATCH setting to limit the size of the request string?

Obviously no is not. THUMBNAILS_BATCH has to be first consumed by the clients to let them decide about the batch size. The error has to be thrown here if client exceeded the batch size.


 UI bit wasn't backported that may be not obvious that first client has to receive